### DISCUSSION TITLE
How do you make a workflow run randomly within a given time period?

### DISCUSSION THREAD

**Discussion by dan-ie** [community/community#131450] [Actions] [answered]
### Select Topic Area

Question

### Body

I want to run a workflow once randomly within a given time period (i. e between 11am and 2pm). How can I do this?

**Comment by grandmas-favorite** [comment, marked as answer]
Hey @k-awn!

While it is supported to schedule a workflow with the native Actions functionality (more on this [here](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows)), there is not an official functionality available for random workflow scheduling.

However, it's possible you could accomplish this by using the GitHub API to trigger a webhook event called [repository_dispatch](https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads#repository_dispatch), which would allow you to trigger a workflow for activity that happens outside of GitHub. 

In other words, you could create your own custom random time generator, and use the API to trigger a workflow based on that.

Hope this helps!
